File: Interview Preparation /5th Problem/Jump Game II.java
class Solution {
    public boolean canJump(int[] nums) {
        int index = nums.length-1;
        for(int i = nums.length - 1;i >= 0;i--){
            int j = nums[i] + i;
            if(j >= index) index = i;
    }
    return index == 0;
    }
}

File: Interview Preparation /5th Problem/Jump Game II.c
int jump(int* nums, int numsSize) {
    int ans = 0;
    int index = 0;
    int far = 0;

    for (int i = 0; i < numsSize - 1; i++) {
        far = fmax(far,(i + nums[i]) );
        if (i == index) {
            ans++;
            index = far;
        }
    }

    return ans;
}
